### PR TITLE
Clean up GLideN64.vcxproj

### PR DESCRIPTION
**Pull request by project64**
Cleaning up the project file, remove a lot of duplication.

have one msvc project instead of two that can be used by any of the recent versions

### CODE REVIEW

**Comment by gonetz** [comment]
Hello zilmar!
Cleanup is always welcome, but there is one major issue with this commit.
When I open that solution in VS 2013, I can't build it because "The build tools for v140 (Platform Toolset = 'v140') cannot be found." I can't use "Upgrade Solution..." as well.
The only option is to edit project files and remove toolset. It is not good.
I need VS 2013 because I often search for regressions. Regressions can be very old, from the years when only project for VS 2013 was available. I can open those project files in recent version of VS, but it will change solution file, thus I had to revert changes after test. That is it will make that long and tedious process even longer.

**Comment by project64** [comment]
Ok mostly I was focusing on GLideN64.vcxproj that one should work in theory, but I agree this makes the pull request incomplete, from your point of view. 

While all the other projects still need to be cleaned up (https://github.com/gonetz/GLideN64/pull/1919/commits/cb8a388f9e4428c73e84ab86ad27ba5a32c925bf) should fix up the other projects to have a lower tool version so they can be opened and to use the default compiler of the visual studio version.

So hopefully with the new change it should work for you in VS 2013

**Comment by gonetz** [comment]
With latest commit I can open solutions in VS 2013 without warning. GLideNUI built successfully, but GLideNHQ can't link because legacy_stdio_definitions.lib is not available for VS 2013. If I remove it from dependencies, the project linked, but now I have missing links when I build GLideN64. So, this project is still incompatible with VS 2013.
I compiled the plugin for Project64 and for mupen64plus with VS 2015 with no issues.

I have plans to drop VS 2013 support after next Public Release and switch to VS 2017. Your work will be very helpful for that. I'll push it to a branch until that.

As an alternative, it is possible to re-work cmake project and build Visual Studio projects with it.
But I'm not that familiar with cmake to do it.

**Comment by gonetz** [comment]
That would be great.

**Comment by project64** [comment]
no problem, I wanted to do more but been busy lately so have not done as much as I wanted. I am going to be busy for the next 3 weeks, then I can hopefully finish what I started. 

things I want to do is look at replacing the UI with WTL instead of QT, and do the testing my self with VS2013, VS2015 and VS2017

**Comment by project64** [comment]
I am going to leave the QT to wtl to another pull request and just try to finish the change. 

the builds at the moment are:
Debug - Used for project64
Release - Used for Project64
Debug_mupen64plus - Used for mupen
Release_mupen64plus - Used for mupen

these are needed, there are 3 other build configurations, are these needed?
MinSizeRel
Release_with_debug_mupen64plus
RelWithDebugInfo

I opened GLideN64.sln with visual studio 2017 and it built Debug_mupen64plus/Win32 and Release_mupen64plus/Win32 with no issue for me .. I do not think I have anything special there, what configuration are you trying to build with VS 2017. I will be looking at VS 2013 shortly

**Comment by gonetz** [comment]
> these are needed, there are 3 other build configurations, are these needed?

I don't use them. Fell free to remove them.

I'm planing to drop VS 2013 support after the next public release. So, if you will not make the project be compatible with VS 2013 it is not a big problem. It is need to be compatible with VS 2015, since I currently work with VS 2015 and it is also used for builds on AppVeyor.

File: src/CommonPluginAPI.cpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
-#ifdef OS_WINDOWS
+#ifdef _WIN32
```
[the remark this answers is not in the document]
Comment (project64) [reply src/CommonPluginAPI.cpp]: OK it is not strictly needed for this pull request. I have started to replace the UI with WTL in some code I have in a folder, but I decided to leave those changes to a future pull request. The problem was that Shlwapi.h has:
#define OS_WINDOWS                  0           // Windows 9x vs. NT

So when I add in the windows headers you get a lot of warnings if I remember correctly which is why I changed it.

**Comment by project64** [comment]
this may be done now

**Comment by theboy181** [comment]
:)

**Comment by project64** [comment]
for Visual Studio 2017 see if this helps (https://stackoverflow.com/questions/38290169/cannot-find-corecrt-h-universalcrt-includepath-is-wrong)

**Comment by project64** [comment]
for Visual Studio 2013 64bit I suspect i need a new freetype.lib built for that environment of I need to be able to include that in the project so that it is built with the same settings

**Comment by gonetz** [comment]
Installation of  "Windows Universal CRT SDK" fixed issues with VS 2017, thanks!
All mupen64plus configurations work flawlessly.
Now I need to build QT for VS 2017.

64bit support for Visual Studio 2013 is not that important, since I use VS2013 mostly to track regressions bugs.

I still need to test Visual Studio 2015 compatibility. Then, I need to setup AppVeyor to use new project and the task will be completed.

zilmar, thanks again for that great work! The project now really looks much more clean and convenient to use.

**Comment by project64** [comment]
I still want to remove the dependency on QT but that will be a new pull request, which will make the build process even simpler

**Comment by project64** [comment]
the compile instructions need to be updated now to but might be worth waiting till after I remove the dependency on qt, cause then it will be even simpler

**Pull request merged**